internal/command/command_service.go

```diff
@@ -193,27 +224,34 @@ func (cs *CommandService) subscribe(ctx context.Context) {
 			return
 		default:
 			subscribeFn := func() error {
-				if subscribeClient == nil {
+				cs.subscribeClientMutex.Lock()
```

Comment:
I'm slightly concerned that we are locking constantly in this subscribe loop. Have we considered a `RWLock` here?

Reply:
Okay didn't use a `RWLock` but moved the locking bit out of the for loop.

Reply:
Need to keep the locking bit in the for loop otherwise the retry logic will fail.
I wouldn't be too concerned about the lock in the for loop as it won't be called that often. The `cs.subscribeClient.Recv()` is a blocking call, so unless we get a lot of messages sent to us in one go we won't be iterating in the for loop that often.